src/snowflake/connector/cursor.py

```diff
@@ -911,6 +926,36 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         """Context manager with commit or rollback."""
         self.close()
 
+    def get_results_from_sfqid(self, sfqid: str):
+        """Gets the results from previously ran query."""
+
+        def wait_until_ready():
+            """Makes sure query has finished executing and once it has retrieves results."""
+            status = self.connection.get_query_status(sfqid)
+            no_data_counter = 0
+            no_data_max_retry = 24
+            while self.connection.is_still_running(status):
+                status = self.connection.get_query_status(sfqid)
+                if not self.connection.is_still_running(status) and status != QueryStatus.SUCCESS:
+                    raise DatabaseError(
+                        "Status of query '{}' is {}, results are unavailable".format(sfqid, status.name))
+                if status == QueryStatus.NO_DATA:
+                    no_data_counter += 1
+                    if no_data_counter > no_data_max_retry:
+                        raise DatabaseError("Cannot retrieve data on the status of this query. No information returned "
```

Comment:
same comment.

Reply:
I have copied this error from the JDBC PR as well.
This would be triggered if someone wanted to wait for results on a still running query, but the query wasn't done in 25 * 5s = 125 s.
This is more to just be consistent with JDBC, maybe @sfc-gh-mknister could also comment if this was specifically requested by someone.

Reply:
Hi man @sfc-gh-mkeller , I was wondering if the pr you referenced was the right one. I guess you might also want to take a look at  https://github.com/snowflakedb/snowflake-jdbc/commit/eaed3362f837c399d95a3e4634a739fc37573fbd. 

So now the code has a problem that we might wait for another 5s even if the async query has finished successfully.  This was also in the original jdbc async pr. The pr I mentioned above is the new one to fix/optimize it, which is also the current async code for JDBC.

Reply:
@sfc-gh-wshangguan I didn't know of this PR. Let me update the retry logic to mimic how JDBC retries.